pkg/gds/certs.go

```diff
@@ -85,7 +85,28 @@ func (s *Service) HandleCertifcateRequests(certDir string) (err error) {
 			wg.Add(1)
 			go func(logctx zerolog.Logger) {
 				defer wg.Done()
-				if err := s.submitCertificateRequest(req); err != nil {
+				// Get the VASP from the certificate request
+				var (
+					vasp *pb.VASP
+					err  error
+				)
+				if vasp, err = s.db.RetrieveVASP(req.Vasp); err != nil {
+					logctx.Error().Str("vasp_id", req.Vasp).Msg("could not retrieve vasp for certificate request submission")
+					return
+				}
+
+				// Verify that the VASP has not errored or been rejected
+				if vasp.VerificationStatus < pb.VerificationState_REVIEWED || vasp.VerificationStatus > pb.VerificationState_VERIFIED {
+					logctx.Info().Str("verification_status", vasp.VerificationStatus.String()).Msg("vasp is not ready for certificate request submission")
+					if err = models.UpdateCertificateRequestStatus(req, models.CertificateRequestState_CR_REJECTED, "certificate request rejected", "automated"); err != nil {
+						logctx.Error().Err(err).Msg("could not update certificate request status")
+						return
+					}
+					if err = s.db.UpdateCertReq(req); err != nil {
+						logctx.Error().Err(err).Msg("could not save updated certificate request")
+						return
+					}
```

Comment:
@pdeziel I still think the `req` is a problem; since the for loop is going to modify `req` on line 75 and so the go routine might have the wrong pointer when it's saving the certificate request; I think the reason we haven't been caught by this bug in the past is that we only are usually doing one cert at a time -- but now that we're doing bulk re-issuance; I'm guessing it's going to crop up.

Reply:
Gotcha, I've updated the PR with that fix.

Reply:
Thanks!